### PR TITLE
Add docs for polygon functions

### LINKED ISSUE #35341: some geo functions are not documented
readWKTMultiPolygon
readWKTPolygon
polygonsWithinSpherical
polygonsDistanceSpherical
polygonsDistanceCartesian
polygonsEqualsCartesian
polygonsSymDifferenceSpherical
polygonsSymDifferenceCartesian
polygonsIntersectionSpherical
polygonsWithinCartesian
polygonConvexHullCartesian
polygonAreaSpherical
polygonsUnionSpherical
polygonPerimeterSpherical
polygonsIntersectionCartesian
polygonAreaCartesian
polygonPerimeterCartesian
polygonsUnionCartesian

https://github.com/ClickHouse/ClickHouse/pull/19257/files
https://github.com/ClickHouse/ClickHouse/blob/master/tests/queries/0_stateless/01302_polygons_distance.sql

### PR DESCRIPTION
**Pull request by DanRoscigno**
Add docs for PR https://github.com/ClickHouse/ClickHouse/pull/19257

closes https://github.com/ClickHouse/ClickHouse/issues/35341

### Changelog category (leave one):
- Documentation (changelog entry is not required)

@nikitamikhaylov can you help with the descriptions for these functions?  I am having a problem with the spherical functions, I do not understand how they work.

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
This is an automated comment for commit 6670cc2154ee0780e5d810a9fc11685339b48e30 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/49512/6670cc2154ee0780e5d810a9fc11685339b48e30/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: docs/en/sql-reference/functions/geo/polygon.md [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+
+### Input parameters
+
+### Returned value
+
+UInt8, 0 for false, 1 for true
+
+## polygonsDistanceSpherical
+
+Calculates
```
[the remark this answers is not in the document]
Comment (nikitamikhaylov) [reply docs/en/sql-reference/functions/geo/polygon.md:77]: `Spherical` means that coordinates are interpreted as a coordinates on a pure and ideal sphere. Which is not true for the Earth. Using this type of coordinate system speeds up execution, but of course is not precise.

[the remark this answers is not in the document]
Comment (nikitamikhaylov) [reply docs/en/sql-reference/functions/geo/polygon.md:77]: https://archive.fosdem.org/2020/schedule/event/working_with_spatial_trajectories_in_boost_geometry/attachments/slides/3988/export/events/attachments/working_with_spatial_trajectories_in_boost_geometry/slides/3988/FOSDEM20_vissarion.pdf

File: docs/en/sql-reference/functions/geo/polygon.md [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+```response
+```
+
+### Input parameters
+
+### Returned value
+
+## polygonAreaSpherical
+
+Calculates
```
Comment (nikitamikhaylov) [inline docs/en/sql-reference/functions/geo/polygon.md:225]: ... the surface area of a polygon.

**Comment by DanRoscigno** [comment]
Thanks so much @nikitamikhaylov. I will go back to the tests and add the rest of the examples and publish this, then add the docs to the source so that they can be maintained there and we have the same info in system.functions and in the published docs.

**Comment by DanRoscigno** [comment]
@nikitamikhaylov I need more help, please.  Can you teach me how to read the code and determine the input params for polygonAreaSpherical()?  It looks like several points and an int.

**Comment by nikitamikhaylov** [comment]
Oh, there is no easy way  to determine the input params. From this line https://github.com/ClickHouse/ClickHouse/blob/c5b521498aa364ab90f76bd6ed4639b6187f7b69/src/Functions/polygonArea.cpp#L52 we understand that the input param count is 1. And from the name of the function we can say that the input param is Polygon, which in fact has type `Array(Array(Tuple(Float64, Float64)))`

**Comment by nikitamikhaylov** [comment]
Some cheat sheet: 

Point is `Tuple(Float64, Float64)`
Ring is `Array(Tuple(Float64, Float64))`
Polygon is `Array(Array(Tuple(Float64, Float64)))`
MultiPolygon is `Array(Array(Array(Tuple(Float64, Float64))))`

**Pull request merged**